implementation/gameengine/common/src/main/java/org/inaetics/dronessimulator/gameengine/common/state/HealthGameEntity.java

```diff
@@ -0,0 +1,38 @@
+package org.inaetics.dronessimulator.gameengine.common.state;
+
+import lombok.EqualsAndHashCode;
+import org.inaetics.dronessimulator.common.D3Vector;
+import org.inaetics.dronessimulator.common.protocol.EntityType;
+
+@EqualsAndHashCode(callSuper=true)
+public abstract class HealthGameEntity<C extends GameEntity<C>> extends GameEntity<C> {
```

Comment:
Yep. C is the type of the child. This generic is usually used in places where you want to define a function that returns the specific type which is inheriting. In this case, it is necassary for the function deepcopy which should return something of the C child.